addons/bestja_detailed_reports/models.py

```diff
@@ -171,6 +198,28 @@ def _compute_project_dates(self):
         return "{} — {}".format(date_start, date_stop)
 
     @api.one
+    @api.depends('project', 'report_entries')
+    def _compute_total_cities_nr(self):
+        """
+            Computes number of unique cities where stores
+            are held, both for the organization and all its children.
+        """
+        project = self.sudo().project
+        level = project.organization.level
+        if level == 1:
+            projects_under = project.children
+            cities_total = []
+            for store in projects_under.stores_in_project:
+                cities_total.append(store.store.city)
+            for store in project.stores_in_project:
+                cities_total.append(store.store.city)
+            tmpset = set(cities_total)
+            cities_total = list(tmpset)
```

Comment:
Yu don't need to cast something to a list to check its length. Sets have lengths also...